Added new neurodamus-core and coreneuron versions and fixed coreneuron tags

iomaganaris opened the pull request.

iomaganaris committed:
Added new coreneuron version and fixed tags

pramodk commented on var/spack/repos/builtin/packages/coreneuron/package.py:
```diff
@@ -37,7 +37,8 @@ class Coreneuron(CMakePackage):
     version('develop', git=url, submodules=True)
-    version('0.14.1', git=url, submodules=True, preferred=True)
+    version('0.15', git=url, tag='0.15', submodules=True)
+    version('0.14', git=url, tag='0.14', submodules=True)
```
@iomaganaris : I think we changed version `0.14.1` with tag `0.14` for some reason...could you check the log of this recipe  change?

iomaganaris replied:
I think we bumped the version there to rebuild the modules with disabled `nrn_timeout`. However, there was not created a new tag in `CoreNeuron`.

iomaganaris committed:
Neurodamus version pinning with neurodamus-core and other modules

- Moved neurodamus-core dependency from neurodamus-model to each neurodamus-<model>
  to be able to change versions of neurodamus core based on the version of the
  specific model
- Made neurodamus-core@2.5.0 the version used for all the versions of
  neurodamus-<model>s up until now
- Pin coreneuron@0.15, neuron@7.6.6, synapsetool@0.5.1 and reportinglib@2.5.1
  with neurodamus-core@2.5.0

matz-e commented:
Don't force push, please? We already squash at the very end.

ferdonline commented on var/spack/repos/builtin/packages/neurodamus-hippocampus/package.py:
```diff
@@ -15,3 +15,7 @@ class NeurodamusHippocampus(NeurodamusModel):
+
+    # NEURODAMUS-HIPPOCAMPUS <=0.3
+    depends_on('neurodamus-core@2.5.0', when='@:0.3')
+    depends_on('neurodamus-core+python@2.5.0', when='+python@:0.3')
```
Do we need to repeat the version for +python? Isn't the first rule always respected?

ferdonline commented on var/spack/repos/builtin/packages/neurodamus-mousify/package.py:
```diff
@@ -14,3 +14,7 @@ class NeurodamusMousify(NeurodamusModel):
+
+    # NEURODAMUS-MOUSIFY <=0.2
+    depends_on('neurodamus-core@2.5.0', when='@:0.2')
+    depends_on('neurodamus-core+python@2.5.0', when='+python@:0.2')
```
idem here and below

ferdonline reviewed (commented):
I'm not very found of the fact that dependencies split over multiple files, and neurodamus-model specifies dependencies based on a dependency version (does it actually work? I did smtg similar in the past that was just ignored!)
I believe these dependencies should move to neurodamus-core as it's neurodamus core which has mod files, and in a perfect world it would provide the partial mech .so files!
So, if this works I find it kind of fragile, but I'm happy to see working tests :)

@pramodk In the future I'd be keep into attempting consolidating neurodamus build (taking advantage of Jorge work to have the single Neuron) and loading the neurodamus and model mechanisms separately. Bonus: Werner will be happy!

pramodk commented:
> @pramodk In the future I'd be keep into attempting consolidating neurodamus build (taking advantage of Jorge work to have the single Neuron) and loading the neurodamus and model mechanisms separately. Bonus: Werner will be happy!

yeah agree....in favour for simplifying stuff!

iomaganaris committed:
Revert "Neurodamus version pinning with neurodamus-core and other modules"

This reverts commit 6da7a5a066324e4b1e67c26c4d0fa9c0c2105776.
It is more appropriate to create a new PR for this change to think and test
it more thoroughly.

iomaganaris committed:
Bumped version of neurodamus-<model>s

pramodk commented on var/spack/repos/builtin/packages/neurodamus-thalamus/package.py:
```diff
@@ -12,5 +12,6 @@ class NeurodamusThalamus(NeurodamusModel):
 
     version('develop', git=git, branch='master', submodules=True, clean=False)
+    version('0.2.1', git=git, tag='0.2', submodules=True, clean=False)
     version('0.2', git=git, tag='0.2', submodules=True, clean=False)
```
In `deploy/configs/applications/modules.yaml` we should only whitelist latest version to avoid all modules?

iomaganaris replied:
Made the required changes in `bbp-packages.yaml`.

matz-e replied:
> In `deploy/configs/applications/modules.yaml` we should only whitelist latest version to avoid all modules?

No. We should whitelist only the desired packages

iomaganaris committed:
Bumped neurodamus-<model>s version in bbp-packages to avoid making
available the old modules

ferdonline committed:
Adding neurodamus core 2.6

ferdonline committed:
Added to deploy

iomaganaris committed:
Changed version formating based on the deployment update

ferdonline commented on var/spack/repos/builtin/packages/neurodamus-thalamus/package.py:
```diff
@@ -12,5 +12,7 @@ class NeurodamusThalamus(NeurodamusModel):
 
     version('develop', git=git, branch='master', submodules=True, clean=False)
+    # Updated CoreNeuron to 0.15 and neurodamus-core to 2.6.0
+    version('0.2-1', git=git, tag='0.2', submodules=True, clean=False)
```
0.2 is not different from 0.2-1, so we can delete it

iomaganaris replied:
I kept it for demonstration of the new versioning system but it should maybe be deleted. You are right :+1:

ferdonline commented:
Maybe this can rebased and go on top of branch "version/neurodamus-core"

iomaganaris committed:
Removed previous versions with same tag

matz-e commented:
force-push again… this is not gerrit :thinking:

iomaganaris commented:
> force-push again… this is not gerrit

I did a rebase to the other branch and then it was needed to do a force push if I am not missing something :confused:

matz-e commented:
> > force-push again… this is not gerrit
> 
> I did a rebase to the other branch and then it was needed to do a force push if I am not missing something confused

Sorry, just got triggered by getting re-routed to a page with nothing on it. Rebase without force push is not possible…

ferdonline commented:
Superseeding with a new PR because we no longer need to create neurodamus redeployment versions since there are actual new versions (due to new random seeding)
